packages/kit/src/compatibility.ts

```diff
@@ -98,19 +98,26 @@ export async function hasNuxtCompatibility (constraints: NuxtCompatibility, nuxt
 }
 
 /**
- * Check if current nuxt instance is version 2 legacy
+ * Check if current Nuxt instance is of specified major version
  */
-export function isNuxt2 (nuxt: Nuxt = useNuxt()) {
+export function isNuxtMajorVersion (majorVersion: 2 | 3 | 4, nuxt: Nuxt = useNuxt()) {
   const version = getNuxtVersion(nuxt)
-  return version[0] === '2' && version[1] === '.'
+
+  return version.startsWith(`${majorVersion}.`)
```

Comment:
Following your advice, let's revert it to a more performance-oriented version! 😄 
```suggestion
  return version[0] === majorVersion.toString() && version[1] === '.'
```